avm/res/service-fabric/cluster/main.bicep

```diff
@@ -29,17 +29,17 @@ param maxUnusedVersionsToKeep int = 3
 @description('Optional. The settings to enable AAD authentication on the cluster.')
 param azureActiveDirectory object = {}
 
-@description('Optional. Describes the certificate details like thumbprint of the primary certificate, thumbprint of the secondary certificate and the local certificate store location.')
-param certificate object = {}
+@description('Conditional. The certificate to use for securing the cluster. The certificate provided will be used for node to node security within the cluster, SSL certificate for cluster management endpoint and default admin client. Required if the certificateCommonNames parameter is not used.')
+param certificate certificateType
 
-@description('Optional. Describes a list of server certificates referenced by common name that are used to secure the cluster.')
-param certificateCommonNames object = {}
+@description('Conditional. Describes a list of server certificates referenced by common name that are used to secure the cluster. Required if the certificate parameter is not used.')
+param certificateCommonNames certificateCommonNamesType
 
 @description('Optional. The list of client certificates referenced by common name that are allowed to manage the cluster.')
-param clientCertificateCommonNames array = []
+param clientCertificateCommonNames clientCertificateCommonNameType
 
 @description('Optional. The list of client certificates referenced by thumbprint that are allowed to manage the cluster.')
-param clientCertificateThumbprints array = []
+param clientCertificateThumbprints clientCertificateThumbprintType
```

Comment:
It looks like this `clientCertificateThumbprints` and `serverCertificateCommonNames` parameters are not validated by any e2e test. I don't want to block the PR on it since it already adds a lot of improvements. However, I'd ask to keep that in mind to be investigated in a next version. If it helps not to forget, feel free to open a GH issue to track the "parameter validation extension" as a future module improvement. You can resolve this comment.

Reply:
Agreed, I added an issue as a reminder to implement.